### PR TITLE
ai: generate: remove endpoint from experiment & remove beta from path

### PR DESCRIPTION
**Pull request by gioelecerati**
**What does this pull request do? Explain your changes. (required)**

- Remove `generate` from experiment
- Remove `beta` from path
- Make sure that the endpoint still works with the deprecated `beta`, to be removed on a second time for a full phase out
- Modified the tests to test both the new and the `beta` paths

**Specific updates (required)**

**How did you test each of these updates (required)**

**Does this pull request close any open issues?**

**Screenshots (optional)**

**Checklist**

- [ X] I have read the **CONTRIBUTING** document.
- [ X] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ X] I have added tests to cover my changes.

### CODE REVIEW

File: packages/api/src/controllers/generate.test.ts [hunk 1]
```diff
@@ -71,6 +71,16 @@ afterEach(async () => {
...
 
+const testBothRoutes = (testFn) => {
+  describe("generate route", () => {
+    testFn("/generate");
+  });
+
+  describe("beta generate route", () => {
+    testFn("/beta/generate");
+  });
+};
```
Comment (victorges) [inline packages/api/src/controllers/generate.test.ts:82]: Keep in mind we need to update CloudFlare routes to also skip the regular worker on the new route.

File: packages/api/src/controllers/generate.test.ts [hunk 2]
```diff
@@ -260,7 +270,7 @@ describe("controllers/generate", () => {
 
     for (const [title, input, error] of testCases) {
       it(title, async () => {
-        const res = await client.fetch("/beta/generate/image-to-image", {
+        const res = await client.fetch("/generate/image-to-image", {
```
Comment (victorges) [inline packages/api/src/controllers/generate.test.ts:273]: Why are these (everything below) not using the base path?

File: packages/api/src/controllers/generate.ts [hunk 3]
```diff
@@ -35,7 +35,13 @@ const aiGenerateDurationMetric = new promclient.Histogram({
...
 const app = Router();
 
-app.use(experimentSubjectsOnly("ai-generate"));
+// TODO: Remove beta paths middleware
+app.use((req, res, next) => {
+  if (req.path.startsWith("/beta/generate")) {
+    req.url = req.url.replace("/beta/generate", "/generate");
+  }
+  next();
+});
```
Comment (victorges) [inline packages/api/src/controllers/generate.ts]: Not sure why do we need this. Aren't the route already registered on both paths?

File: packages/api/src/controllers/generate.ts [hunk 4]
```diff
@@ -35,7 +35,13 @@ const aiGenerateDurationMetric = new promclient.Histogram({
 
 const app = Router();
 
-app.use(experimentSubjectsOnly("ai-generate"));
```
Comment (victorges) [inline packages/api/src/controllers/generate.ts:38]: Yeah this is fine to remove .

File: packages/api/src/schema/pull-ai-schema.js [hunk 5]
```diff
@@ -60,10 +60,20 @@ const downloadAiSchema = async () => {
...
 
+  // TODO: Remove beta paths
+  const newPaths = {};
+  Object.entries(schema.paths).forEach(([path, value]) => {
+    const generatePath = `/api/generate${path}`;
+    const betaGeneratePath = `/api/beta/generate${path}`;
+    newPaths[generatePath] = value;
+    newPaths[betaGeneratePath] = value;
+  });
+  schema.paths = newPaths;
```
Comment (victorges) [inline packages/api/src/schema/pull-ai-schema.js]: I don't think we should have both in the API schema. It should be just the new one. Meaning we only change L76 (prev L66) in this file.

Also, it seems like you updated this script but didn't run it? The current changes on `ai-api-schema` do not seem to correspond to the changes you made in this file. Meaning, in the current code, the `ai-api-schema` should have both the `/beta/` and non`-/beta/` APIs. But instead, it looks like you manually updated the APIs to strip the `/beta/`?

FTR that file should NEVER be updated manually. More info in the schemas README.

Comment (gioelecerati) [reply packages/api/src/schema/pull-ai-schema.js]: Oh, right, I updated it manually when it should have been via script, removing the old path here and fixing the schema

File: packages/api/src/controllers/generate.test.ts (hunk 2 above)
Comment (gioelecerati) [reply packages/api/src/controllers/generate.test.ts:273]: I wanted to just test the new path working, considering we will remove the `beta` endpoint at some point, but yeah I can add it to all of them

File: packages/api/src/controllers/generate.ts (hunk 3 above)
Comment (gioelecerati) [reply packages/api/src/controllers/generate.ts]: Right, this is not needed

File: packages/api/src/controllers/generate.test.ts (hunk 2 above)
Comment (victorges) [reply packages/api/src/controllers/generate.test.ts:273]: Yeah if we have some of the tests testing both, let's be consistent. I'm fine with just testing the new one as well and leaving the old untested. But we can do that when we remove the `/beta`/ ones

File: packages/api/src/schema/ai-api-schema.yaml [hunk 6]
```diff
@@ -368,6 +368,65 @@ paths:
               schema:
                 $ref: '#/components/schemas/studio-api-error'
       x-speakeasy-name-override: segmentAnything2
+  /api/generate/llm:
```
Comment (victorges) [inline packages/api/src/schema/ai-api-schema.yaml:371]: Keep in mind you might have conflict with @mjh1's #2319, but it should be solved by simply running the `pull-ai-schema` script again

**Review by victorges — APPROVED** [review]
LGTM

**Pull request merged**